lib/salus/scanners/report_go_dep.rb

```diff
@@ -7,38 +7,131 @@
 module Salus::Scanners
   class ReportGoDep < Base
     def run
-      unless should_run?
+      if @repository.go_sum_present?
+        record_dep_from_go_sum
+      elsif @repository.go_mod_present?
+        record_dep_from_go_mod
+      elsif @repository.dep_lock_present?
+        record_dep_from_go_lock_package
+      else
         raise(
           InvalidScannerInvocationError,
-          'Cannot report on Go dependencies without a Gopkg.lock file.'
+          'Cannot report on Go dependencies without a Gopkg.lock, go.mod, or go.sum file'
         )
       end
+    end
+
+    def record_dep_from_go_sum
+      go_sum_path = "#{@repository.path_to_repo}/go.sum"
+      dep_list = []
+
+      File.foreach(go_sum_path).each("=\n") do |line|
+        next if line.empty?
```

Comment:
What about lines with other types whitespaces, like as pre/suffixes?
I think you should use String#strip before checking if line is empty
https://ruby-doc.org/core-3.0.2/String.html#method-i-strip

Same for your other record_ methods.

Reply:
Changed 👍